{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "main", "filename": "understanding_tasks.py", "msg": "Main function calling"}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "task_factory", "filename": "tasks.py", "msg": {"msg": "initalizing task", "vars": {"loop": "<ProactorEventLoop running=True closed=False debug=True>", "coro": "<coroutine object first at 0x00000246daf48100>", "task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ initalized"}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "self.__step scheduled on EVENT_LOOP in __init__", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ after scheduling __step on EVENT_LOOP using loop.call_soon", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "Entered self.__step", "method_signature": "__step(self, exc = None)", "vars": {"task_name": "Custom-task-Task-1", "exc": "None", "sent_from": "'__init__'"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "will initalized self._fut_waiter value", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "initalized self._fut_waiter to None", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. Call to __step_run_and_handle_request in next line.", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_request entered", "method_signature": "__step_run_and_handle_result(self, exc)", "vars": {"task_name": "Custom-task-Task-1", "exc": "None"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "`__step_run_and_hanle_result` main try-except exc is None so `coro.send()` in next line.", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "first", "filename": "understanding_tasks.py", "msg": "======== FIRST.FIRST ==================="}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg:": "__step_run_and_handle_result main try-except after `coro.send()`", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "try-else in __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1", "result": "<Future pending created at C:\\msys64\\mingw64\\lib\\python3.12\\asyncio\\base_events.py:448>", "is_asyncio_future_blocking": "getattr(result,\"_asyncio_future_blocking\", None)= True"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "is_future blocking", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1", "is_future_blocking": true}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "scheduling on EVENT_LOOP callback to fututre.add_done_call_back", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1", "result": "<Future pending created at C:\\msys64\\mingw64\\lib\\python3.12\\asyncio\\base_events.py:448>"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "After scheduling callbacks to fut. self._future_wait = result", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1", "result": "<Future pending cb=[Task.__wakeup()] created at C:\\msys64\\mingw64\\lib\\python3.12\\asyncio\\base_events.py:448>", "self._fut_waiter": null, "future_callbacks_list(fut._callbacks)": "[(<bound method Task.__wakeup of <Task pending name='Custom-task-Task-1' coro=<first() running at c:/Users/shaik/learning/concurrency/python_concurrency/understanding_tasks.py:12> cb=[gather.<locals>._done_callback() at C:\\msys64\\mingw64\\lib\\python3.12\\asyncio\\tasks.py:767] created at c:\\Users\\shaik\\learning\\concurrency\\python_concurrency\\experiments\\tasks.py:410>>, <_contextvars.Context object at 0x00000246daf85980>)]"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self._fut_waiter set to result", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1", "self._fut_waiter": "<Future pending cb=[Task.__wakeup()] created at C:\\msys64\\mingw64\\lib\\python3.12\\asyncio\\base_events.py:448>"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self finally set to None", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. After __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`. Call to _leave_task next and then set self to None", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:35 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`.Custom-task-Task-1 set to none", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__wakeup", "filename": "tasks.py", "msg": {"msg": "__wakeup entered", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__wakeup", "filename": "tasks.py", "msg": {"msg": "will call future.result()", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__wakeup", "filename": "tasks.py", "msg": {"msg": "__wakeup after future.result()", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1", "future._state": "FINISHED"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__wakeup", "filename": "tasks.py", "msg": {"msg": "__wakeup not issue when called. calling again __step with no exc. I guess StopIteration will be raised", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "Entered self.__step", "method_signature": "__step(self, exc = None)", "vars": {"task_name": "Custom-task-Task-1", "exc": "None", "sent_from": "'_wakeup._else'"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "will initalized self._fut_waiter value", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "initalized self._fut_waiter to None", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. Call to __step_run_and_handle_request in next line.", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_request entered", "method_signature": "__step_run_and_handle_result(self, exc)", "vars": {"task_name": "Custom-task-Task-1", "exc": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "`__step_run_and_hanle_result` main try-except exc is None so `coro.send()` in next line.", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "first", "filename": "understanding_tasks.py", "msg": "========== END ==========="}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_result StopIteration encountered but the task is not cancelled.", "vars": {"task_name": "Custom-task-Task-1", "exc.value": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "super().set_result(exc.value)", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self finally set to None", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. After __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`. Call to _leave_task next and then set self to None", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`.Custom-task-Task-1 set to none", "vars": {"task_name": "Custom-task-Task-1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__wakeup", "filename": "tasks.py", "msg": {"msg": "end of __wakeup. set self = None", "vars": {"task_name": "Custom-task-Task-1", "future_name": "Custom-task-Task-1: name: 1"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "task_factory", "filename": "tasks.py", "msg": {"msg": "initalizing task", "vars": {"loop": "<ProactorEventLoop running=False closed=False debug=True>", "coro": "<coroutine object BaseEventLoop.shutdown_asyncgens at 0x00000246daf94040>", "task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "task_factory", "filename": "tasks.py", "msg": {"msg": "initalizing task", "vars": {"loop": "<ProactorEventLoop running=False closed=False debug=True>", "coro": "<coroutine object BaseEventLoop.shutdown_asyncgens at 0x00000246daf94040>", "task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ initalized"}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ initalized"}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "self.__step scheduled on EVENT_LOOP in __init__", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "self.__step scheduled on EVENT_LOOP in __init__", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ after scheduling __step on EVENT_LOOP using loop.call_soon", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ after scheduling __step on EVENT_LOOP using loop.call_soon", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "Entered self.__step", "method_signature": "__step(self, exc = None)", "vars": {"task_name": "Custom-task-Task-2", "exc": "None", "sent_from": "'__init__'"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "Entered self.__step", "method_signature": "__step(self, exc = None)", "vars": {"task_name": "Custom-task-Task-2", "exc": "None", "sent_from": "'__init__'"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "will initalized self._fut_waiter value", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "will initalized self._fut_waiter value", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "initalized self._fut_waiter to None", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "initalized self._fut_waiter to None", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. Call to __step_run_and_handle_request in next line.", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. Call to __step_run_and_handle_request in next line.", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_request entered", "method_signature": "__step_run_and_handle_result(self, exc)", "vars": {"task_name": "Custom-task-Task-2", "exc": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_request entered", "method_signature": "__step_run_and_handle_result(self, exc)", "vars": {"task_name": "Custom-task-Task-2", "exc": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "`__step_run_and_hanle_result` main try-except exc is None so `coro.send()` in next line.", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "`__step_run_and_hanle_result` main try-except exc is None so `coro.send()` in next line.", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_result StopIteration encountered but the task is not cancelled.", "vars": {"task_name": "Custom-task-Task-2", "exc.value": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_result StopIteration encountered but the task is not cancelled.", "vars": {"task_name": "Custom-task-Task-2", "exc.value": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "super().set_result(exc.value)", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "super().set_result(exc.value)", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self finally set to None", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self finally set to None", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. After __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. After __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`. Call to _leave_task next and then set self to None", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`. Call to _leave_task next and then set self to None", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`.Custom-task-Task-2 set to none", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`.Custom-task-Task-2 set to none", "vars": {"task_name": "Custom-task-Task-2"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "task_factory", "filename": "tasks.py", "msg": {"msg": "initalizing task", "vars": {"loop": "<ProactorEventLoop running=False closed=False debug=True>", "coro": "<coroutine object BaseEventLoop.shutdown_default_executor at 0x00000246daefee40>", "task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "task_factory", "filename": "tasks.py", "msg": {"msg": "initalizing task", "vars": {"loop": "<ProactorEventLoop running=False closed=False debug=True>", "coro": "<coroutine object BaseEventLoop.shutdown_default_executor at 0x00000246daefee40>", "task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "task_factory", "filename": "tasks.py", "msg": {"msg": "initalizing task", "vars": {"loop": "<ProactorEventLoop running=False closed=False debug=True>", "coro": "<coroutine object BaseEventLoop.shutdown_default_executor at 0x00000246daefee40>", "task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ initalized"}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ initalized"}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ initalized"}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "self.__step scheduled on EVENT_LOOP in __init__", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "self.__step scheduled on EVENT_LOOP in __init__", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "self.__step scheduled on EVENT_LOOP in __init__", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ after scheduling __step on EVENT_LOOP using loop.call_soon", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ after scheduling __step on EVENT_LOOP using loop.call_soon", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__init__", "filename": "tasks.py", "msg": {"msg": "__init__ after scheduling __step on EVENT_LOOP using loop.call_soon", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "Entered self.__step", "method_signature": "__step(self, exc = None)", "vars": {"task_name": "Custom-task-Task-3", "exc": "None", "sent_from": "'__init__'"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "Entered self.__step", "method_signature": "__step(self, exc = None)", "vars": {"task_name": "Custom-task-Task-3", "exc": "None", "sent_from": "'__init__'"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "Entered self.__step", "method_signature": "__step(self, exc = None)", "vars": {"task_name": "Custom-task-Task-3", "exc": "None", "sent_from": "'__init__'"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "will initalized self._fut_waiter value", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "will initalized self._fut_waiter value", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "will initalized self._fut_waiter value", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "initalized self._fut_waiter to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "initalized self._fut_waiter to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "initalized self._fut_waiter to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. Call to __step_run_and_handle_request in next line.", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. Call to __step_run_and_handle_request in next line.", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. Call to __step_run_and_handle_request in next line.", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_request entered", "method_signature": "__step_run_and_handle_result(self, exc)", "vars": {"task_name": "Custom-task-Task-3", "exc": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_request entered", "method_signature": "__step_run_and_handle_result(self, exc)", "vars": {"task_name": "Custom-task-Task-3", "exc": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_request entered", "method_signature": "__step_run_and_handle_result(self, exc)", "vars": {"task_name": "Custom-task-Task-3", "exc": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "`__step_run_and_hanle_result` main try-except exc is None so `coro.send()` in next line.", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "`__step_run_and_hanle_result` main try-except exc is None so `coro.send()` in next line.", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "`__step_run_and_hanle_result` main try-except exc is None so `coro.send()` in next line.", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_result StopIteration encountered but the task is not cancelled.", "vars": {"task_name": "Custom-task-Task-3", "exc.value": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_result StopIteration encountered but the task is not cancelled.", "vars": {"task_name": "Custom-task-Task-3", "exc.value": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "__step_run_and_handle_result StopIteration encountered but the task is not cancelled.", "vars": {"task_name": "Custom-task-Task-3", "exc.value": "None"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "super().set_result(exc.value)", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "super().set_result(exc.value)", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "super().set_result(exc.value)", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self finally set to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self finally set to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step_run_and_handle_result", "filename": "tasks.py", "msg": {"msg": "self finally set to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. After __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. After __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`try` of try-finally in `self.__step`. After __step_run_and_handle_result", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`. Call to _leave_task next and then set self to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`. Call to _leave_task next and then set self to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`. Call to _leave_task next and then set self to None", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`.Custom-task-Task-3 set to none", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`.Custom-task-Task-3 set to none", "vars": {"task_name": "Custom-task-Task-3"}}}
{"datetime": "2025-07-30 12:04:39 AM", "level": "INFO", "func_name": "__step", "filename": "tasks.py", "msg": {"msg": "`finally` of try-finally in `self.__step`.Custom-task-Task-3 set to none", "vars": {"task_name": "Custom-task-Task-3"}}}
